tests/test_celery.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+from django.apps import apps
+from django.conf import settings
+from mock import patch
+
+from health_check.contrib.celery.apps import HealthCheckConfig
+from health_check.contrib.celery.backends import CeleryHealthCheck
+
+
+class TestCeleryHealthCheck:
+    HEALTH_CHECK_CLASS = CeleryHealthCheck
+    CELERY_APP_CONTROL_PING = (
+        "health_check.contrib.celery.backends.app.control.ping"
+    )
+    CELERY_APP_CONTROL_INSPECT_ACTIVE_QUEUES = (
+        "health_check.contrib.celery.backends.app.control.inspect.active_queues"
+    )
+
+    def test_check_status_doesnt_add_errors_when_ping_successfull(self):
+        with patch(
+            self.CELERY_APP_CONTROL_PING,
+            return_value=[{"celery@4cc150a7b49b": {"ok": "pong"}}],
+        ), patch(
+            self.CELERY_APP_CONTROL_INSPECT_ACTIVE_QUEUES,
+            return_value={
+                "celery@4cc150a7b49b": [
+                    {"name": queue.name} for queue in settings.CELERY_QUEUES
+                ]
+            },
+        ):
+            health_check = self.HEALTH_CHECK_CLASS()
+            health_check.check_status()
+
+            assert len(health_check.errors) == 0
+
+    def test_check_status_reports_errors_if_ping_responses_are_incorrect(self):
+        with patch(
+            self.CELERY_APP_CONTROL_PING,
+            return_value=[
+                {
+                    "celery1@4cc150a7b49b": {"ok": "pong"},
+                    "celery2@4cc150a7b49b": {},
+                    "celery3@4cc150a7b49b": {"error": "pong"},
+                }
+            ],
+        ):
+            health_check = self.HEALTH_CHECK_CLASS()
+            health_check.check_status()
+
+            assert len(health_check.errors) == 2
+
+    def test_check_status_adds_errors_when_ping_successfull_but_not_all_defined_queues_have_consumers(
+        self,
+    ):
+        queues = settings.CELERY_QUEUES
```

Comment:
`.pop()` is used later in this test on a mutable object, so let's create a shallow copy of `settings.CELERY_QUEUES`:

```suggestion
        queues = list(settings.CELERY_QUEUES)
```